embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/execution/OkHttpRequestExecutionService.kt

```diff
@@ -0,0 +1,98 @@
+package io.embrace.android.embracesdk.internal.delivery.execution
+
+import io.embrace.android.embracesdk.internal.comms.api.ApiRequestV2
+import io.embrace.android.embracesdk.internal.comms.api.ApiResponse
+import io.embrace.android.embracesdk.internal.comms.api.Endpoint
+import io.embrace.android.embracesdk.internal.delivery.SupportedEnvelopeType
+import okhttp3.Headers.Companion.toHeaders
+import okhttp3.OkHttpClient
+import okhttp3.Protocol
+import okhttp3.Request
+import okhttp3.RequestBody
+import okio.BufferedSink
+import okio.buffer
+import okio.source
+import java.io.InputStream
+import java.net.HttpURLConnection
+import java.util.concurrent.TimeUnit
+
+private const val NO_HTTP_RESPONSE = -1
+private const val TOO_MANY_REQUESTS = 429
+private const val DEFAULT_TIMEOUT_SECONDS = 60L
+
+class OkHttpRequestExecutionService(
+    private val coreBaseUrl: String,
+    private val lazyDeviceId: Lazy<String>,
+    private val appId: String,
+    private val embraceVersionName: String,
+) : RequestExecutionService {
+
+    private val okHttpClient = OkHttpClient()
+        .newBuilder()
+        .connectTimeout(DEFAULT_TIMEOUT_SECONDS, TimeUnit.SECONDS)
```

Comment:
At 60s, this seems to be too long. What is the default? 15s? We should probably just use that.

If we can't establish a TCP connection in even 15s, it's unlikely to succeed unless the network connection changes.

Reply:
That's what we were setting with OkHttpUrlConnection, but I agree, the default will probably be enough

Reply:
yikes. yeah, this could leave the network request hanging and increase the failure rate of requests that are queued but not sent, so ultimately not serialized and redelivered....

Reply:
Lets leave the old one with these bad values, but fix OkHttp to have reasonable ones. It matter less now with the new delivery layer, but the read timeout at 60 may be too long too, but we can keep that as is.